### PR TITLE
Add --normalize flag for Java

### PR DESCRIPTION
**Pull request by mbrdl**
Adresses #1028. I'm not too happy with the way language-specific options work right now but I guess that's a topic for another day.

### CODE REVIEW

**Comment by mbrdl** [comment]
I would like to adjust the test but I don't know how.

**Comment by mbrdl** [comment]
Thanks, where can I view it? Your JPlag fork isn't public

**Comment by mbrdl** [comment]
You can do that? I can't find it

**Comment by mbrdl** [comment]
Looks good, thanks 👍

**Comment by mbrdl** [comment]
But maybe normalization shouldn't be a language-specific option at all, since in principle any language can support it. Maybe it should just be a global option with a disclaimer. idk

**Comment by tsaglam** [comment]
@mbrdl we still have some open requests here.

File: language-api/src/main/java/de/jplag/options/DummyOption.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package de.jplag.options;
+
+class DummyOption<T> implements LanguageOption<T> {
```
Comment (tsaglam) [inline language-api/src/main/java/de/jplag/options/DummyOption.java:3]: Javadoc to explain the purpose.

File: language-api/src/main/java/de/jplag/options/LanguageOptions.java [hunk 2]
```diff
@@ -16,8 +18,19 @@ public abstract class LanguageOptions {
     /**
      * New instance
      */
-    protected LanguageOptions() {
+    public LanguageOptions(boolean supportsNormalization) {
         this.options = new ArrayList<>();
+        if (supportsNormalization) {
+            var description = "Enable token string normalization, which can help detect certain cases of plagiarism.";
```
Comment (tsaglam) [inline language-api/src/main/java/de/jplag/options/LanguageOptions.java:24]: Avoid var, especially when String is nearly just as short.

**Comment by mbrdl** [comment]
Putting this off for now due to unresolved design decisions and lack of urgency

**Comment by mbrdl** [comment]
todo: add cli tests for the flag

File: language-api/src/main/java/de/jplag/options/LanguageOptions.java [hunk 3]
```diff
@@ -16,8 +18,19 @@ public abstract class LanguageOptions {
...
+            this.normalize = createDefaultOption(OptionType.bool(), "normalize", description, false);
+        }
+    }
+
+    /**
+     * New instance
+     */
```
Comment (dfuchss) [inline language-api/src/main/java/de/jplag/options/LanguageOptions.java:31]: Document default option

**Review by dfuchss — COMMENTED** [review]
Minor comment

**Comment by dfuchss** [comment]
@mbrdl this branch has some merge conflicts; could you resolve them :)

**Comment by mbrdl** [comment]
This PR is on ice for now, we'll probably implement the option globally instead

**Pull request closed without merging**